osquery/filesystem/filesystem.cpp

```diff
@@ -276,14 +276,43 @@ static void genGlobs(std::string path,
                      GlobLimits limits) {
   // Use our helped escape/replace for wildcards.
   replaceGlobWildcards(path, limits);
+  // inodes of directory symlinks for loop detection
+  std::set<int> dsym_inos;
 
   // Generate a glob set and recurse for double star.
-  size_t glob_index = 0;
-  while (++glob_index < kMaxRecursiveGlobs) {
+  for (size_t glob_index = 0; ++glob_index < kMaxRecursiveGlobs;) {
     auto glob_results = platformGlob(path);
 
-    for (auto const& result_path : glob_results) {
+    for (auto& result_path : glob_results) {
       results.push_back(result_path);
+
+#ifndef WIN32
```

Comment:
NB: try to reduce the amount of code that exists on one platform but not others. "tighter" if/else in the preprocessor is better.

Reply:
Yeah, I don't know why I I updated the diff with this. This obviously shouldn't be merged.